network-mux/src/Network/Mux/Types.hs

```diff
@@ -150,6 +154,14 @@ instance (MiniProtocolLimits ptcl) => MiniProtocolLimits (MiniProtocolId ptcl) w
     maximumIngressQueue DeltaQ     = 0xffff
     maximumIngressQueue (AppProtocolId pid) = maximumIngressQueue pid
 
+-- | Control Interface for the Initiator side of Miniprotocols.
+-- The outer STM action is used to signal that the miniprotocol should start.
+-- The inner STM action provides the result of the miniprotocol.
+newtype MiniProtocolInitiatorControl m a = MiniProtocolInitiatorControl (STM m (STM m a))
+
+-- | STM action for reading the result of the responder side of a Miniprotocol.
+newtype MiniProtocolResponderControl m b = MiniProtocolResponderControl (STM m b)
+
```

Comment:
It would be easier to read this patch if these types where directly in `Mux.hs`.  They are internal so no need to expose them here.

Reply:
They are not internal to `Mux.hs`. They are part of the mux interface, see Interface.hs.

Reply:
Still I would put them in `Mux.hs` and then we can export them from `Mux.Interface`.

At some point we'll need to refactor how things are exported, `Network.Mux` ought to be the interface module and `Network.Mux.Muxxer` or `.Internal` could have the implementation.

Reply:
There is a PR for refactoring Mux, #1247 which moves declarations around.
This PR conforms to how types are declared right now.